diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 5c06604..54450b6 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -23,11 +23,23 @@ add_executable(spherical options.cpp spherical.cpp)
 target_link_libraries(spherical PUBLIC siscone_spherical)
 
 if (MSVC)
-  find_package(unofficial-getopt-win32 REQUIRED)
-  target_link_libraries(siscone_main PRIVATE unofficial::getopt-win32::getopt)
-  target_link_libraries(siscone_area PRIVATE unofficial::getopt-win32::getopt)
-  target_link_libraries(siscone_test PRIVATE unofficial::getopt-win32::getopt)
-  target_link_libraries(times PRIVATE unofficial::getopt-win32::getopt)
-  target_link_libraries(sample PRIVATE unofficial::getopt-win32::getopt)
-  target_link_libraries(spherical PRIVATE unofficial::getopt-win32::getopt)
-endif()
\ No newline at end of file
+  add_library(getopt SHARED IMPORTED)
+  set_target_properties(getopt PROPERTIES
+                          IMPORTED_IMPLIB "$ENV{BUILD_PREFIX}\\Library\\lib\\getopt.lib"
+                          IMPORTED_LOCATION "$ENV{BUILD_PREFIX}\\Library\\bin\\getopt.dll"
+                       )
+
+  target_include_directories(siscone_main PRIVATE "$ENV{BUILD_PREFIX}\\Library\\include")
+  target_include_directories(siscone_area PRIVATE "$ENV{BUILD_PREFIX}\\Library\\include")
+  target_include_directories(siscone_test PRIVATE "$ENV{BUILD_PREFIX}\\Library\\include")
+  target_include_directories(times PRIVATE "$ENV{BUILD_PREFIX}\\Library\\include")
+  target_include_directories(sample PRIVATE "$ENV{BUILD_PREFIX}\\Library\\include")
+  target_include_directories(spherical PRIVATE "$ENV{BUILD_PREFIX}\\Library\\include")
+
+  target_link_libraries(siscone_main PRIVATE getopt)
+  target_link_libraries(siscone_area PRIVATE getopt)
+  target_link_libraries(siscone_test PRIVATE getopt)
+  target_link_libraries(times PRIVATE getopt)
+  target_link_libraries(sample PRIVATE getopt)
+  target_link_libraries(spherical PRIVATE getopt)
+endif()
