--- runtime/src/js/pgadmin.js
+++ runtime/src/js/pgadmin.js
@@ -62,7 +62,11 @@
 // Override the paths above, if a developer needs to
-if (fs.existsSync('dev_config.json')) {
-  try {
-    let dev_config = JSON.parse(fs.readFileSync('dev_config.json'));
-    pythonPath = path.resolve(dev_config['pythonPath']);
-    pgadminFile = path.resolve(dev_config['pgadminFile']);
+const configFilePath = path.join(__dirname, 'dev_config.json');
+if (fs.existsSync(configFilePath)) {
+  try {
+    const configFileContent = fs.readFileSync(configFilePath, 'utf8'); // Specify encoding
+    let dev_config = JSON.parse(configFileContent);
+    const condaPrefix = path.normalize(process.env['CONDA_PREFIX']);
+    pythonPath = path.resolve(condaPrefix, process.env['PGADMIN4_PY_EXEC']);
+    pgadminFile = path.resolve(condaPrefix, process.env['PGADMIN4_PY_HOME']);
+    console.warn('PREFIX: ' + condaPrefix + '\nPGADMIN4_PY_EXEC: ' + process.env['PGADMIN4_PY_EXEC'] + '\nPGADMIN4_PY_HOME: ' + process.env['PGADMIN4_PY_HOME'] + '\npythonPath: ' + pythonPath + '\npgadminFile: ' + pgadminFile);
   } catch (error) {
