From 7d929c0b553156f6beb8003dc0790fc78e32c35f Mon Sep 17 00:00:00 2001
From: Daniel Ching <dching@nvidia.com>
Date: Thu, 29 May 2025 15:31:35 -0500
Subject: [PATCH 7/9] BLD: Use shared libtar instead of static

---
 cmake/Dependencies.common.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/Dependencies.common.cmake b/cmake/Dependencies.common.cmake
index b0af9097..85582936 100644
--- a/cmake/Dependencies.common.cmake
+++ b/cmake/Dependencies.common.cmake
@@ -116,7 +116,7 @@ endif()
 ##################################################################
 if(BUILD_LIBTAR)
   find_library(libtar_LIBS
-          NAMES libtar.a tar libtar
+          NAMES tar libtar
           PATHS ${LIBTAR_ROOT_DIR} "/usr/local" ${CMAKE_SYSTEM_PREFIX_PATH}
           PATH_SUFFIXES lib lib64)
   if(${libtar_LIBS} STREQUAL libtar_LIBS-NOTFOUND)
-- 
2.34.1

