From 149e79fb7b3dc87a71a3e302b5f4d75750259d1f Mon Sep 17 00:00:00 2001
From: Adam Ehlers Nyholm Thomsen <aen@atp.dk>
Date: Mon, 5 Aug 2024 14:59:43 +0200
Subject: [PATCH] Got error about missing std::tolower on MSVC

---
 highs/util/stringutil.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/highs/util/stringutil.cpp b/highs/util/stringutil.cpp
index 49bceaa0f..b5700d226 100644
--- a/highs/util/stringutil.cpp
+++ b/highs/util/stringutil.cpp
@@ -9,6 +9,7 @@
 
 #include <algorithm>  // for std::transform
 #include <cassert>
+#include <cctype>
 
 /*
 void strRemoveWhitespace(char* str) {
