From 6e04abc6eabb944fa23b462c4e2c308fd92dc9ef Mon Sep 17 00:00:00 2001
From: Tobias Fischer <info@tobiasfischer.info>
Date: Sun, 8 Sep 2024 13:46:28 +1000
Subject: [PATCH] Bump pyo3 version for py313 compatibility

---
 python/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/Cargo.toml b/python/Cargo.toml
index 8e8ccb4..92ebbad 100644
--- a/python/Cargo.toml
+++ b/python/Cargo.toml
@@ -9,9 +9,9 @@ crate-type = ["cdylib"]
 
 [dependencies]
 neuromorphic-drivers = "0.13.1"
-numpy = "0.21.0"
+numpy = "0.22.0"
 paste = "1.0"
-pyo3 = {version = "0.21", features = ["extension-module"]}
+pyo3 = {version = "0.22", features = ["extension-module"]}
 
 [build-dependencies]
 cc = "1.0"
diff --git a/python/src/lib.rs b/python/src/lib.rs
index 76b7bd5..2b1b0e5 100644
--- a/python/src/lib.rs
+++ b/python/src/lib.rs
@@ -4,6 +4,7 @@ mod structured_array;
 extern crate neuromorphic_drivers as neuromorphic_drivers_rs;
 use pyo3::IntoPy;
 use std::ops::DerefMut;
+use pyo3::types::PyModuleMethods;
 
 type ListedDevice = (String, String, Option<String>, Option<String>);
 
