diff --git a/setup.py b/setup.py
index b58ec5a..a99ba38 100644
--- a/setup.py
+++ b/setup.py
@@ -11,5 +11,5 @@ setup(
     packages=['modAL', 'modAL.models', 'modAL.utils'],
     classifiers=['Development Status :: 4 - Beta'],
     install_requires=['numpy', 'scikit-learn>=0.18',
-                      'scipy>=0.18', 'pandas>=1.1.0', 'skorch==0.9.0'],
+                      'scipy>=0.18', 'pandas>=1.1.0', 'skorch>=0.9.0'],
 )
