diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed194c68..eb3c2d6b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -188,6 +188,7 @@ option(FASTJET_HAVE_EXPLICIT_FOR_OPERATORS  "Compiler supports explicit operator
 
 # support for swig-based python bindings
 option(FASTJET_ENABLE_PYTHON "Enable building of swig-based fastjet python bindings" OFF)
+option(FASTJET_PYTHON_ONLY   "Only build the swig-based fastjet python bindings"     OFF)
 
 #-------------------------------------------------------------------------
 # compile-time options
@@ -269,7 +270,7 @@ message(STATUS "FastJet compiler options: ${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_$
 #-------------------------------------------------------------------------
 # python Support (off by default)
 #
-if (FASTJET_ENABLE_PYTHON)
+if (FASTJET_ENABLE_PYTHON OR FASTJET_PYTHON_ONLY)
   find_package(SWIG 3.0 COMPONENTS python REQUIRED)
 endif()
 set(FASTJET_PYTHON_PACKAGE_NAME "fastjet" CACHE STRING "The name of the installed fastjet python package.")
@@ -336,23 +337,36 @@ set(CONFIG_SUMMARY "FastJet configuration summary:
 
 #-------------------------------------------------------------------------
 # get into directories:
-add_subdirectory(include)  # for basic FJ headers
-add_subdirectory(src)      # for the main FJ library
-add_subdirectory(tools)    # fastjet toolkit
-add_subdirectory(plugins)  # fastjet clustering plugins
+if (NOT FASTJET_PYTHON_ONLY)
+  add_subdirectory(include)  # for basic FJ headers
+  add_subdirectory(src)      # for the main FJ library
+  add_subdirectory(tools)    # fastjet toolkit
+  add_subdirectory(plugins)  # fastjet clustering plugins
+endif()
 message(STATUS "Installation prefix is ${CMAKE_INSTALL_PREFIX}")
-if (FASTJET_ENABLE_PYTHON)
+if (FASTJET_ENABLE_PYTHON OR FASTJET_PYTHON_ONLY)
+  if (FASTJET_PYTHON_ONLY)
+    message("FINDING FASTJET FOR PYTHON-ONLY BUILD")
+    find_package(fastjet REQUIRED)
+    if (fastjet_FOUND)
+      message("${fastjet_DIR}")
+    endif()
+  endif()
   add_subdirectory(pyinterface) # for python interface
   if (NOT IS_ABSOLUTE "${FASTJET_PYTHON_INSTALL_DIR}")
     set(FASTJET_PYTHON_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/${FASTJET_PYTHON_INSTALL_DIR}")
   endif()
   message(STATUS "Python interface enabled and will be installed to ${FASTJET_PYTHON_INSTALL_DIR}")
 endif()
-add_subdirectory(example)  # for examples (gets built after all the rest)
+if (NOT FASTJET_PYTHON_ONLY)
+  add_subdirectory(example)  # for examples (gets built after all the rest)
+endif()
 
 
 # generate the configure file
@@ -389,36 +405,37 @@ add_test(
 include(CMakePackageConfigHelpers)
 
 # allow Fastjet to work with find_package
-export(EXPORT FastjetTargets
-  NAMESPACE fastjet::
-  FILE "${CMAKE_CURRENT_BINARY_DIR}/fastjetTargets.cmake"
-)
-
-install(EXPORT FastjetTargets
-  NAMESPACE fastjet::
-  FILE fastjetTargets.cmake
-  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/fastjet
-)
-
-configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/Config.cmake.in
-  "${CMAKE_CURRENT_BINARY_DIR}/fastjetConfig.cmake"
-  INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/fastjet
-  NO_SET_AND_CHECK_MACRO
-  NO_CHECK_REQUIRED_COMPONENTS_MACRO
-  )
-
-write_basic_package_version_file(
-  "${CMAKE_CURRENT_BINARY_DIR}/fastjetConfigVersion.cmake"
-  VERSION "${PROJECT_VERSION}"
-  COMPATIBILITY AnyNewerVersion
-  )
-
-install(FILES
-  ${CMAKE_CURRENT_BINARY_DIR}/fastjetConfig.cmake
-  ${CMAKE_CURRENT_BINARY_DIR}/fastjetConfigVersion.cmake
-  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/fastjet
-  )
-
+if (NOT FASTJET_PYTHON_ONLY)
+  export(EXPORT FastjetTargets
+    NAMESPACE fastjet::
+    FILE "${CMAKE_CURRENT_BINARY_DIR}/fastjetTargets.cmake"
+    )
+
+  install(EXPORT FastjetTargets
+    NAMESPACE fastjet::
+    FILE fastjetTargets.cmake
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/fastjet
+    )
+
+  configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/Config.cmake.in
+    "${CMAKE_CURRENT_BINARY_DIR}/fastjetConfig.cmake"
+    INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/fastjet
+    NO_SET_AND_CHECK_MACRO
+    NO_CHECK_REQUIRED_COMPONENTS_MACRO
+    )
+
+  write_basic_package_version_file(
+    "${CMAKE_CURRENT_BINARY_DIR}/fastjetConfigVersion.cmake"
+    VERSION "${PROJECT_VERSION}"
+    COMPATIBILITY AnyNewerVersion
+    )
+
+  install(FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/fastjetConfig.cmake
+    ${CMAKE_CURRENT_BINARY_DIR}/fastjetConfigVersion.cmake
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/fastjet
+    )
+endif()
 
 message(STATUS "${CONFIG_SUMMARY}")
 
diff --git a/pyinterface/CMakeLists.txt b/pyinterface/CMakeLists.txt
index 7016f179..632786fe 100644
--- a/pyinterface/CMakeLists.txt
+++ b/pyinterface/CMakeLists.txt
@@ -16,6 +16,7 @@ set_property(SOURCE fastjet.i PROPERTY SWIG_MODULE_NAME __init__)
 
 swig_add_library(fastjet_swig LANGUAGE python SOURCES fastjet.i OUTPUT_DIR fastjet_swig)
 set_property(TARGET fastjet_swig PROPERTY SWIG_USE_TARGET_INCLUDE_DIRECTORIES TRUE)
+
 #set_property(TARGET fastjet_swig PROPERTY INSTALL_RPATH lib)
 #set_property(TARGET fastjet_swig PROPERTY INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
 
@@ -29,8 +30,8 @@ endif()
 # Unix only, Windows is different
 set_property(TARGET fastjet_swig PROPERTY SUFFIX ".${Python_SOABI}${CMAKE_SHARED_MODULE_SUFFIX}")
 
-target_link_libraries(fastjet_swig PUBLIC fastjet)
-target_link_libraries(fastjet_swig PUBLIC fastjettools)
+target_link_libraries(fastjet_swig PUBLIC fastjet::fastjet)
+target_link_libraries(fastjet_swig PUBLIC fastjet::fastjettools)
 target_link_libraries(fastjet_swig PUBLIC Python::Module)
 
 #set_target_properties(fastjet_swig PROPERTIES INSTALL_RPATH "$ORIGIN/../../../../lib")
