From d3d9392a4d95b86ed487e77d9ad1f01861e5d259 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Sat, 24 May 2025 23:10:25 +0200
Subject: [PATCH] Remove -fhardened from compiler flags

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b202263..fa6de0c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -415,7 +415,7 @@ if (ENABLE_ASAN OR ENABLE_UBSAN)
 endif()
 
 # try to ensure some compiler sanity and hardening options where supported
-foreach (flag -fno-strict-overflow -fno-delete-null-pointer-checks -fhardened)
+foreach (flag -fno-strict-overflow -fno-delete-null-pointer-checks)
 	check_c_compiler_flag(${flag} found)
 	if (found)
 		add_compile_options(${flag})
-- 
2.49.0

